climada/hazard/tc_tracks.py

```diff
@@ -1249,12 +1258,70 @@ def from_netcdf(cls, folder_name):
                                "whole life time.")
                 basin = track.basin
                 del track.attrs['basin']
-                track['basin'] = ("time", np.full(track.time.size, basin))
+                track['basin'] = ("time", np.full(track.time.size, basin, dtype="<U2"))
             data.append(track)
         tr = cls()
         tr.data = data
         return tr
 
+    def write_hdf(self, file_name, complevel=4):
+        """Write TC tracks in HDF5 format.
+
+        Parameters
+        ----------
+        file_name: str or Path
+            Path to a new HDF5 file. If it exists already, the file is overwritten.
+        complevel : int
+            Specifies a compression level (0-9) for the gzip compression of the data.
+            A value of 0 or None disables compression. Default: 4
+        """
+        with h5py.File(file_name, 'w') as store:
+            store.create_dataset('size', (), dtype=int, data=self.size)
+
+            LOGGER.info('Writing %d tracks to %s', self.size, file_name)
+            last_perc = 0
+            for i, tr in enumerate(self.data):
+                perc = 100 * i / self.size
+                if perc - last_perc >= 10:
+                    LOGGER.info("Progress: %d%%", perc)
+                    last_perc = perc
+
+                tr['basin'] = tr['basin'].astype('<U2')
```

Comment:
Do tracks only have basin as a non-data attribute? Would it not be safer to make a loop in case extra attributes are set (would of course only work for certain attribute types though)...

Reply:
No, this is a misunderstanding. This is just changing the dtype to make sure the data is compatible with NetCDF3. In my latest commit, I mention this in an inline comment.